Ajout des commandes de développeurs

Le-Roux-nard opened the pull request:
Ajout de la commande "Execute as Member"

Techniquement fonctionnel mais chaque commande doit être adaptée pour pouvoir defer ou non en fonction de si l'interaction est déjà répondue

Le-Roux-nard committed:
add dev command (execute as) and force fetch guild on start up to ensure name is defined

Le-Roux-nard committed:
Merge branch 'master' of https://github.com/Roux2D2-Co/WaifuBot into dev_functions

Le-Roux-nard committed:
ajout méthode pour récupérer les options nécéssaires à la simulation de n'importe quelle commande (hormis importation de fichiers)

Le-Roux-nard committed:
execute as fonctionnel

Le-Roux-nard commented:
Je dois retravailler la commande pour permettre l'intégration des commandes avec autocomplétion

Le-Roux-nard committed:
Merge branch 'master' of https://github.com/Roux2D2-Co/WaifuBot into dev_functions

Le-Roux-nard committed:
update commands to handle being 'executed as' and using an already answered interaction

Grizfreak reviewed (dismissed):
Checked between the lines, looks good

Le-Roux-nard commented on src/interactions/Commands/ChatInputCommands/devFunctions.ts:
```diff
@@ -0,0 +1,444 @@
+	) as ApplicationCommandSubCommandData;
+
+	const commandOptions = [];
+	if (!!subCommand) {
+		if (!!subCommandGroup) {
+			commandOptions.push({
+				name: subCommandGroup.name,
+				type: subCommandGroup.type,
+				options: [
+					{
+						name: subCommand.name,
+						type: subCommand.type,
+						options: formattedOptions,
+					},
+				],
+			});
+		} else {
+			commandOptions.push({
+				name: subCommand.name,
+				type: subCommand.type,
+				options: formattedOptions,
+			});
+		}
+	} else {
+		commandOptions.push(...formattedOptions);
+	}
```
Waiting for refactorisation :
Add "subCommandOptions" variable and if subCommandGroup condition is true then wrap subCommandOptions inside another variable

```typescript
if (!!subCommand) {
        // if it's a subcommand we're storing the payload we will be using in case it needs to be reused by subCommandGroup and avoid code repetition
        const subCommandOptions = {
		name: subCommand.name,
		type: subCommand.type,
		options: formattedOptions,
	}
	if (!!subCommandGroup) {
                // if there is a subcommandGroup we encase what we prepared before in a subcommandGroup data format
		subCommandOptions = {
			name: subCommandGroup.name,
			type: subCommandGroup.type,
			options: [subCommandOptions]
		}
	} 
        commandOptions.push(subCommandOptions);
} else {
	commandOptions.push(...formattedOptions);
}
```

Le-Roux-nard commented on src/interactions/Commands/ChatInputCommands/devFunctions.ts:
```diff
@@ -0,0 +1,444 @@
+				)
+			);
+		} else if (option.type === ApplicationCommandOptionType.Channel) {
+			components.push(
+				new ActionRowBuilder<ChannelSelectMenuBuilder>().addComponents(
+					new ChannelSelectMenuBuilder().setCustomId(option.name).setPlaceholder(option.name).setMaxValues(1).setMinValues(1)
+				)
+			);
```
Add potential channel filter from option in the channel select menu

Le-Roux-nard reviewed (commented):
There is still work to do

Le-Roux-nard committed:
update types

Le-Roux-nard committed:
Merge branch 'master' of https://github.com/Roux2D2-Co/WaifuBot into dev_functions

Le-Roux-nard committed:
Typo in customButton class type property

Le-Roux-nard committed:
update packages

Le-Roux-nard committed:
add func to split arr in chunks

Le-Roux-nard committed:
update packages + fix type related errors

Le-Roux-nard committed:
Suppression fichier test

Le-Roux-nard committed:
Changement de style du bouton STOP

Le-Roux-nard committed:
lint + assure la transparence de l'image du trade

Le-Roux-nard committed:
Suppression réponse lorsque drop validé

Le-Roux-nard committed:
typage et logs revus

Le-Roux-nard committed:
Modification de la classe CustomButton

Modification apportées dans le but de supporter plus facilement l'ajout ou le retrait de certaines propriétés
(Mise en commentaire du code précédent au cas où)

Le-Roux-nard committed:
Déplacement de la déclaration d'AnilistWaifu d'une interface à un type

Passage de la déclaration d'AnilistWaifu en type plutôt qu'avec une interface
Ajout de spécifications sur ce qui est renvoyé en fonction du contexte

Suppression de l'ancien fichier et modification des imports là où cela était nécessaire

Le-Roux-nard committed:
Correction paramètre not_in lors d'un getRandomCharacter

Correction d'un oubli suite à un débug, cela forçait le paramètre not_in à une valeur fixe et permettait donc de roll un personnage déjà obtenu

Le-Roux-nard committed:
Modification de l'import pour supporter un nombre de waifus plus importants + linter

Le-Roux-nard committed:
Ajout fichier conf debugger VSC

Ajout du fichier de configuration pour le débugger de Visual Studio Code

EIdras commented on src/classes/AnilistWaifu.d.ts:
```diff
@@ -0,0 +1,60 @@
+	image: WaifuImage;
+	name: WaifuName;
+	gender: string;
+	media: If<FromAnilist, { edges: WaifuMediaEdges<FromAnilist> }, { nodes: WaifuMediaNodes }>;
```
Remove FromAnilist

Le-Roux-nard commented on src/interactions/Commands/ChatInputCommands/devFunctions.ts:
```diff
@@ -0,0 +1,476 @@
+		if (!commands) return interaction.respond([]);
+		else {
+			let commandNames = commands
+				.map((command: { name: string }) => {
```
passer à un flatMap

Le-Roux-nard commented on src/interactions/Commands/ChatInputCommands/devFunctions.ts:
```diff
@@ -0,0 +1,476 @@
+
+	async onAutocomplete(interaction: AutocompleteInteraction) {
+		let focusedInteraction = interaction.options.getFocused(true);
+		let value = focusedInteraction.value;
```
rename "value" to make it more understandable that we are talking of current user input

EIdras commented on src/interactions/Commands/ChatInputCommands/devFunctions.ts:
Add access control 
\+ comments and refactoring

Le-Roux-nard committed:
Improved devFunctions and "Execute As"

Le-Roux-nard committed:
Update Waifu queries and types

Update waifus related types and queries to revert the "FromAnilist" subtype change
Fix typo in Roll command
Revert deleteUnknownCommands to true
Add isAdult media property to waifu database type

EIdras reviewed (approved):
![](https://media1.tenor.com/m/CDsybQkGp_kAAAAC/thumbs-up-approve.gif)

Grizfreak reviewed (commented):
Nicely done
